internal/dag/secret.go

```diff
@@ -50,35 +51,47 @@ func isValidSecret(secret *v1.Secret) (bool, error) {
 }
 
 func validateCertificate(data []byte) error {
-	cert, err := decodePEM(data)
-	if err != nil {
-		return err
-	}
-	_, err = x509.ParseCertificate(cert.Bytes)
-	return err
-}
-
-func validatePrivateKey(data []byte) error {
-	key, err := decodePEM(data)
-	if err != nil {
-		return nil
-	}
-	if _, err := x509.ParsePKCS1PrivateKey(key.Bytes); err == nil {
-		return nil
-	}
-	if _, err := x509.ParsePKCS8PrivateKey(key.Bytes); err == nil {
-		return nil
+	var exists bool
+	var block *pem.Block
+	for rest := data[:]; len(rest) > 0; {
```

Comment:
rest := data[:] is a no-op, it copies the value data into a rest, but data is itself a copy of the value passed to validateCertificate, all three point to the same underlying array albeit with their own values for Len and cap.

This code is fine, but I'm worried that a casual reader will assume that data and rest are more independent than they are; I think

for len(data)> 0 {
    block, data := per.Decode(data)
    ...
}

would probably be less open to misinterpretation.

Reply:
i meant to clean that up after i was done delve'ing :) its not a nit :)

Reply:
I meant to drop that after i was done delv'ing.  That's my bad.